tests/test_s3_bucket_operations.py

```diff
@@ -0,0 +1,189 @@
+import logging
+import os
+from datetime import datetime, timedelta, timezone
+
+import pytest
+
+from noobaa_sa.exceptions import BucketAlreadyExists, BucketNotEmpty, NoSuchBucket
+
+log = logging.getLogger(__name__)
+
+
+class TestS3BucketOperations:
+    """
+    Test S3 bucket operations using NSFS
+    """
+
+    def test_bucket_creation_deletion_and_head(self, c_scope_s3client):
```

Comment:
Can we have head_bucket as separate operation in same test

Reply:
Per our discussion, the new implementation of this makes use of the return codes of head_bucket as well so it's sufficient - resolving this conversation